_ingest-pipelines/processors/script.md

```diff
@@ -0,0 +1,155 @@
+`lang`  | Optional  | The programming language of the script. Default is `painless`.
+`params` | Optional |  The parameters that can be passed to the script.
+`description`  | Optional  | A description of the processor's purpose or configuration.
+`if` | Optional | Specifies to conditionally execute the processor.
+`ignore_failure` | Optional | Specifies to ignore failures for the processor. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
+`on_failure` | Optional | Specifies a list of processors to run if the processor fails during execution. These processors are executed in the order they are specified. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
+`tag` | Optional | An identifier tag for the processor. Useful for debugging in order to distinguish between processors of the same type.
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline named `my-script-pipeline` that uses the `script` processor to convert the `message` field to uppercase: 
+
+```json
+PUT _ingest/pipeline/my-script-pipeline
+{
+  "description": "Example pipeline using the ScriptProcessor",
+  "processors": [
+    {
+      "script": {
+        "source": "ctx.message = ctx.message.toUpperCase()",
+        "lang": "painless",
+        "description": "Convert message field to uppercase"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 2 (Optional): Test the pipeline
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/my-script-pipeline/_simulate
+{
+  "docs": [
+    {
+      "_source": {
+        "message": "hello, world!"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "message": "HELLO, WORLD!"
+        },
+        "_ingest": {
+          "timestamp": "2024-05-30T16:24:23.30265405Z"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 3: Ingest a document 
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+POST testindex1/_doc?pipeline=my-script-pipeline
+{
+  "message": "hello, world!"
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+This response confirms that the document is indexed into `testindex1` and is indexing all documents with the `message` field converted to uppercase.
```

Comment:
"is indexing" => either "has indexed" or "will index"

Reply:
Revised "has indexed"